src/radau.jl

```diff
@@ -0,0 +1,351 @@
+            for l = 1:dof
+                for k = 1:dof
+                    AoplusI2[(i-1)*stageNum+l,(j-1)*stageNum+k] =btab.a[i,j]*I[l,k]
+                end
+            end
+        end
+    end
+
+    #IoplusM = [I[i,j]*M for i=1:stageNum, j=1:stageNum]
+    IoplusM=zeros(stageNum*dof,stageNum*dof)
+    for i=1:stageNum
+        for j = 1:stageNum
+            for l = 1:dof
+                for k = 1:dof
+                    IoplusM[(i-1)*stageNum+l,(j-1)*stageNum+k] =I[i,j]*M[l,k]
+                end
+            end
+        end
+    end
+
+    G =  IoplusM-h*AoplusJ
+    Ginv = inv(G)
+
+    # Use Netwon interation
+    #
+    #   ⃗z^(k+1) = ⃗z ^ (k) - Δ⃗z^(k)
+    #TODO: use the transformation T^(-1)A^(-1)T = Λ, W^k = (T^-1 ⊕ I)Z^k
+    ## initial variables iteration
+    #TODO: use better initial values for zpre
+    #w = hnew/hpre
+    #zpre = q(w)+ypre-y
+    zpre = zeros(dof)
+    Δzpre
+    κ
+
+    iterate = true
+    count = 0
+    while iterate
+        Δz = reshape(Ginv*[(-zpre + h*AoplusI2*F(f,z,y,t,c,h))...],dof,stageNum)
+        z = zpre + Δz
+
+        # Stop condition for the Netwon iteration
+        if (count >=1)
+            Θ = norm(Δz)/norm(Δzpre)
+            η = Θ/(1-Θ)
+            if η*norm(Δz) <= κ*min(reltol,abstol)
+                iterate = false
+                break
+            end
+        end
+
+        zpre = z
+        Δzpre = Δz
+
+        count+=1
+    end
+
+    # Once Netwon method converges after some m steps, estimated next step size
+    #
+    #   y = ypre + h ∑b_i*k_i^{m}
+    #
+
+    d = b*inv(A)
+    ynext = ypre
+    for i = 1 : stageNum
+        ynext += z[i]*d[i]
+    end
+end
+
+function errorcontrol!(st)
+    @unpack st:M, h, A, J, tpre, ypre, b̂, b, c, g, order_number, fac
+    γ0 = filter(λ -> imag(λ)==0, eig(inv(A)))
+
+    for i = 1:order_number
+        sum_part += (b̂[i] - b[i]) * h *f(xpre + c[i] * h, g[i])
+    end
+
+    yhat_y = γ0 * h * f(tpre, ypre) + sum_part
+    err = norm( inv(M - h * λ * J) * (yhat_y) )
+
+    hnew = fac * h * err_norm^(-1/4)
+
+    # Update state
+    st.hnew = hnew
+
+    return err, Nothing, st
+end
+
+function ordercontrol!(st)
+    @unpack st:W, step, order
+
+    if step < 10
+        # update state
+        st.order = 5
+
+    else
+        θk = norm(W[  ]) / norm(W[  ])
+        θk_1 = norm(W[  ]) / norm(W[  ])
+
+        Φ_k = √(θk * θk_1)
+    end
+
+end
+###########################################
+# Other help functions
+###########################################
+function constRadauTableau(stageNum)
+    # Calculate c_i, which are the zeros of
+    #              s-1
+    #            d       s-1        s
+    #           ---    (x    * (x-1)  )
+    #              s-1
+    #           dx
+    roots = Array(Float64, stageNum - 1)
+    append!(roots, [1 for i= 1:stageNum])
+    poly = Polynomials.poly([roots;])
+    for i = 1 : stageNum-1
+        poly = Polynomials.polyder(poly)
+    end
+    C = Polynomials.roots(poly)
```

Comment:
Polynomials package has been 3 years didn't update, and Polynomials package has serious performance and the returning type issues. Maybe need to rewrite this part.